### PR TITLE
Unify gdynet and dynet use in DyNet

### PR DESCRIPTION
**Pull request by xunzhang**
- The remaining stuff for this:
- [x] Address `examples/python/cpu_vs_gpu.py`
- [x] Address https://github.com/clab/dynet/pull/731
- [x] Document

- Usage:
`python model.py`: only cpu will be used in both backends by default
`python model.py --dynet-gpu`: will use `GPU:0`
`python model.py --dynet-devices GPU:1`: will use specified device(s)
`python model.py --dynet-gpus 1`: will use 1 gpu device
If you want to change the default behavior of `python model.py`, you can use `dynet_config` in your `model.py` script like this:
```
import dynet_config
dynet_config.set_gpu()
dynet_config.set(mem=4,random_seed=9)
import dynet as dy
```

- Summary: remove `import _gdynet` functionality, unify this with `import dynet` together with the input arguments.

### CODE REVIEW

**Comment by neubig** [comment]
Thanks a lot! I'll take a look. FYI: the C++ documentation may have to be updated as well.

**Comment by xunzhang** [comment]
cc @neubig

File: doc/source/install.rst [hunk 1]
```diff
@@ -109,18 +109,12 @@ GPU (CUDA) support
 
 DyNet supports running programs on GPUs with CUDA. If you have CUDA
 installed, you can build DyNet with GPU support by adding
-``-DBACKEND=cuda`` to your cmake options. This will result in three
-libraries named "libdynet" and "libgdynet" being
-created. When you want to run a program on CPU, you can link to the
-"libdynet" library as shown above. When you want to run a program on
-GPU, you can link to the "libgdynet" library.
+``-DBACKEND=cuda`` to your cmake options. The linking method is exact
```
Comment (neubig) [inline doc/source/install.rst]: "exactly the same as with the CPU backend case"

File: python/setup.py.in [hunk 2]
```diff
@@ -0,0 +1,96 @@
+from setuptools import setup
```
Comment (neubig) [inline python/setup.py.in]: Perhaps this file can be deleted with the new changes to `setup.py`?

**Pull request merged**